src/config.rs

```diff
@@ -273,6 +303,16 @@ fn parse_default<T: FromStr>(env: &str, default: T) -> Result<T, Error> {
     parse(env).map(|v| v.unwrap_or(default))
 }
 
+fn parse_duration(env: &str) -> Result<Option<Duration>, Error> {
+    parse::<String>(env)?
+        .map(|ds| duration_str::parse(&ds).map_err(|_| Error::EnvVar(env.to_string(), ds)))
+        .transpose()
+}
+
+fn parse_duration_default(env: &str, default: Duration) -> Result<Duration, Error> {
```

Comment:
This seems infallible? Why return a result?

Reply:
parse_duration returns Result<Option>. It may fail to parse, for example if the var is `x` that is not a valid. We default if its unset, not if it is invalid

Reply:
ty